PaulasWeatherApp/PaulasWeatherApp/Model/Singleton.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  Singleton.swift
+//  PaulasWeatherApp
+//
+//  Created by Stef, Paula on 8/1/22.
+//
+
+import Foundation
+import UIKit
+
+class Singleton: NSObject {
+    
+    static let sharedInstance = Singleton()
+    var hourlyWeather: [WeatherDetails] = []
+    
+    func resizeImage(image: UIImage, targetSize: CGSize) -> UIImage {
+       let size = image.size
+       
+       let widthRatio  = targetSize.width  / size.width
+       let heightRatio = targetSize.height / size.height
+       
+       // Figure out what our orientation is, and use that to form the rectangle
+       var newSize: CGSize
+       if(widthRatio > heightRatio) {
+           newSize = CGSize(width: size.width * heightRatio, height: size.height * heightRatio)
+       } else {
+           newSize = CGSize(width: size.width * widthRatio,  height: size.height * widthRatio)
```

Comment:
So you need to use as multiply value the minimum between height and width ratio? instead of having an if, you can do something like: 
let ratio = min(widthRatio, heightRatio)
newSize = CGSize(width: size.width * ratio,  height: size.height * ratio)